### PR TITLE
Added Google group support

### PR DESCRIPTION
**Pull request by micahhausler**
For your reviewing pleasure, I broke it up into 2 commits.

### CODE REVIEW

File: cmd/server.go [hunk 1]
```diff
@@ -70,10 +72,28 @@ func serverFunc(cmd *cobra.Command, args []string) {
...
+	ud, err := groupauth.NewUserDirectory(
+		data,
+		viper.GetString("admin-username"),
+		viper.GetString("domain"),
+		viper.GetStringSlice("groups"),
+		viper.GetString("redis-host"),
+		viper.GetDuration("validity"),
+	)
+	if err != nil {
+		zap.L().Fatal("", zap.Error(err))
```
Comment (mrhwick) [inline cmd/server.go]: May want to add a message here

**Review by mrhwick — APPROVED** [review]

File: cmd/server.go [hunk 2]
```diff
@@ -70,11 +72,29 @@ func serverFunc(cmd *cobra.Command, args []string) {
 		zap.L().Info("Metrics server gracefully stopped")
 	}()
 
+	// Group authorization initialization
+	data, err := ioutil.ReadFile(viper.GetString("service-account"))
```
Comment (ethanfrogers) [inline cmd/server.go]: small nit; it may be better to give this a more meaningful name.

File: cmd/server.go [hunk 3]
```diff
@@ -70,11 +72,29 @@ func serverFunc(cmd *cobra.Command, args []string) {
...
+	if err != nil {
+		zap.L().Fatal("Error creating directory", zap.Error(err))
+	}
+
 	authMux := http.NewServeMux()
 	authMux.HandleFunc("/api/v1/sign", authority.SignHTTP)
 	handler := middlewares.Apply(
 		authMux,
-		oidcauth.OidcUserContext(
+		ud.Authorize(oidcauth.FromContext),
```
Comment (ethanfrogers) [inline cmd/server.go:96]: Does this not need to be flipped around?

**Review by ethanfrogers — APPROVED** [review]
LGTM. I forgot to submit my review last night.

File: cmd/server.go (hunk 3 above)
Comment (micahhausler) [reply cmd/server.go:96]: No, it is a bit counterintuitive, but it is the correct order.

**Pull request merged**